### ISSUE TITLE
[BUG][Kotlin] "date-time" type integrity violated when using default dateLibrary (java8)

### ISSUE THREAD

**Issue by alixwar** [OpenAPITools/openapi-generator#20078]
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [ ] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

When generating the domain model using default configuration of dateLibrary (=java8 according to https://openapi-generator.tech/docs/generators/kotlin/) the generated date-time objects are represented by java.time.OffsetDateTime.
These objects are evaluated to strings that look like this:

````
2024-11-08T06:54:06.831187+01:00
```

Date times should follow the format of RFC 3339 according to https://swagger.io/docs/specification/v3_0/data-models/data-types/#strings 

In other words they should look like this:
```
2017-07-21T17:32:28Z
```

Workaround: Specify another date library, such as kotlinx-datetime which represents date-times with kotlinx.datetime.Instant

##### openapi-generator version

7.9.0

##### OpenAPI declaration file content or url

```yaml
        foo:
          type: string
          format: 'date-time'
          description: Example
```

##### Generation Details

Generated using Gradle plugin

##### Steps to reproduce

```
gradlew openApiGenerate
```

##### Related issues/PRs


##### Suggest a fix
